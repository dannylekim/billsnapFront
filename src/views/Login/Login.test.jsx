import React from 'react';
import ReactDOM from 'react-dom';
import LoginPage from './index.js';

describe('LoginPage', () => {
  let wrapper;
  let handleMockFunction;

  beforeEach(() => {
    handleMockFunction = jest.fn();
    wrapper = shallow(<LoginPage />);
  })

  afterEach(() => {
    handleMockFunction.mockRestore();
  })

  describe('render', () => {
    it('should match snap shot', () => {
      // matches(<LoginPage />);
      expect(wrapper).toMatchSnapshot()
    });
  });

  describe('functions', () => {
    it('should call handler function when GenericButton is click', () => {
      console.log(wrapper.html());
      wrapper.find('.billsnap__generic-btn').simulate('click');
      jest.spyOn(wrapper.instance().handleButtonClick).mockImplementation(handleMockFunction);
      expect(handleMockFunction).toHaveBeenCalled();
    });
  });
});
