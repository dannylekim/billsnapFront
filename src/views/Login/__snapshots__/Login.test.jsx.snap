// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage render should match snap shot 1`] = `
<div
  className="page__login"
>
  <div
    className="login__container"
  >
    <p>
      Maybe something should go here...
    </p>
    <button
      className="billsnap__generic-btn"
      onClick={[Function]}
    >
      Login button
    </button>
  </div>
</div>
`;
